flair/nn/model.py

```diff
@@ -434,71 +434,66 @@ def evaluate(
+                    classification_report_dict["micro avg"][precision_recall_f1] = classification_report_dict[
+                        "accuracy"
+                    ]
+
+            detailed_result = (
+                "\nResults:"
+                f"\n- F-score (micro) {round(classification_report_dict['micro avg']['f1-score'], 4)}"
+                f"\n- F-score (macro) {round(classification_report_dict['macro avg']['f1-score'], 4)}"
+                f"\n- Accuracy {accuracy_score}"
+                "\n\nBy class:\n" + classification_report
             )
-            accuracy_score = micro_f_score = macro_f_score = main_score = 0.0
-            classification_report = ""
-            classification_report_dict = {}
-
-        detailed_result = (
-            "\nResults:"
-            f"\n- F-score (micro) {micro_f_score}"
-            f"\n- F-score (macro) {macro_f_score}"
-            f"\n- Accuracy {accuracy_score}"
-            "\n\nBy class:\n" + classification_report
-        )
 
-        scores: Dict[Union[Tuple[str, ...], str], Any] = {}
+            # Create and populate score object for logging with all evaluation values, plus the loss
+            scores: Dict[Union[Tuple[str, ...], str], Any] = {}
 
-        for avg_type in ("micro avg", "macro avg"):
-            for metric_type in ("f1-score", "precision", "recall"):
-                if avg_type == "micro avg" and avg_type not in classification_report_dict:
-                    value = classification_report_dict["accuracy"]
+            for avg_type in ("micro avg", "macro avg"):
+                for metric_type in ("f1-score", "precision", "recall"):
+                    scores[(avg_type, metric_type)] = classification_report_dict[avg_type][metric_type]
 
-                else:
-                    value = classification_report_dict[avg_type][metric_type]
+            scores["accuracy"] = accuracy_score
 
-                scores[(avg_type, metric_type)] = value
+            if average_over > 0:
+                eval_loss /= average_over
+            scores["loss"] = eval_loss.item()
 
-        scores["accuracy"] = accuracy_score
-
-        if average_over > 0:
-            eval_loss /= average_over
-        scores["loss"] = eval_loss.item()
+            return Result(
+                main_score=classification_report_dict[main_evaluation_metric[0]][main_evaluation_metric[1]],
+                detailed_results=detailed_result,
+                classification_report=classification_report_dict,
+                scores=scores,
+            )
 
-        result = Result(
-            main_score=main_score,
-            detailed_results=detailed_result,
-            classification_report=classification_report_dict,
-            scores=scores,
-        )
+        else:
+            # issue error and default all evaluation numbers to 0.
+            error_text = (
+                f"ACHTUNG! It was not possible to compute evaluation values because: "
```

Comment:
I think this PR is a good opportunity to remove the German word and phrase it completely in english.
Besides that:
The current output looks like this:
```
2023-11-13 11:07:43,133 ACHTUNG! It was not possible to compute evaluation values because: - The evaluation data has no gold labels for label_type='nel'!- And no predictions were made!Double check your corpus (if the test split has labels), and how you initialize the ModelTrainer!
2023-11-13 11:07:43,133 ACHTUNG! It was not possible to compute evaluation values because: - The evaluation data has no gold labels for label_type='nel'!- And no predictions were made!Double check your corpus (if the test split has labels), and how you initialize the ModelTrainer!
```
I would
* Fix the spacing
* Not log it twice: either log the error or add it in the detailed_results, but not both

Reply:
Fixed in latest push.